### PR TITLE
Fixed waf building on OSX

### PR DESCRIPTION
**Pull request by joshdekock**
I've fixed building on OSX using waf, and I have setup a base to start replacing deprecated APIs (set up objc++ compilation). 
Build passing on both linux & OSX: https://travis-ci.org/joshdekock/jack2/builds/97066539

### CODE REVIEW

**Comment by sletz** [comment]
Which "deprecated APIs" ? 

Le 15 d�c. 2015 � 21:13, Josh de Kock <notifications@github.com> a �crit :

**Comment by joshdekock** [comment]
GetMacOSStatusErrorString() from ea8e853

And there are many more to be fixed (excerpt): 
```
../macosx/coreaudio/JackCoreAudioDriver.h:175:13: warning: 'Jack::JackCoreAudioDriver::Open' hides overloaded virtual function [-Woverloaded-virtual]
        int Open(jack_nframes_t buffer_size,
            ^
/Users/ethe/jack2/common/JackAudioDriver.h:69:21: note: hidden overloaded virtual function 'Jack::JackAudioDriver::Open' declared here: different number of parameters (11 vs 20)
        virtual int Open(jack_nframes_t buffer_size,
                    ^
../macosx/coreaudio/JackCoreAudioDriver.mm:139:11: warning: 'AudioHardwareGetPropertyInfo' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetPropertyInfo(kAudioHardwarePropertyDevices, &size, &isWritable);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:318:1: note: 'AudioHardwareGetPropertyInfo' has been explicitly marked deprecated here
AudioHardwareGetPropertyInfo(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:147:11: warning: 'AudioHardwareGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetProperty(kAudioHardwarePropertyDevices, &size, devices);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:339:1: note: 'AudioHardwareGetProperty' has been explicitly marked deprecated here
AudioHardwareGetProperty(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:156:15: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
        err = AudioDeviceGetProperty(devices[i], 0, false, kAudioDevicePropertyDeviceName, &size, device_name_aux);
              ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:177:11: warning: 'AudioHardwareGetPropertyInfo' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetPropertyInfo(kAudioHardwarePropertyDevices, &size, &isWritable);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:318:1: note: 'AudioHardwareGetPropertyInfo' has been explicitly marked deprecated here
AudioHardwareGetPropertyInfo(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:185:11: warning: 'AudioHardwareGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetProperty(kAudioHardwarePropertyDevices, &size, devices);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:339:1: note: 'AudioHardwareGetProperty' has been explicitly marked deprecated here
AudioHardwareGetProperty(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:207:11: warning: 'AudioHardwareGetPropertyInfo' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetPropertyInfo(kAudioHardwarePropertyDevices, &size, &isWritable);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:318:1: note: 'AudioHardwareGetPropertyInfo' has been explicitly marked deprecated here
AudioHardwareGetPropertyInfo(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:215:11: warning: 'AudioHardwareGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    err = AudioHardwareGetProperty(kAudioHardwarePropertyDevices, &size, devices);
          ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:339:1: note: 'AudioHardwareGetProperty' has been explicitly marked deprecated here
AudioHardwareGetProperty(   AudioHardwarePropertyID inPropertyID,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:226:15: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
        err = AudioDeviceGetProperty(devices[i], 0, false, kAudioDevicePropertyDeviceUID, &size, &UIname);
              ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:234:15: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
        err = AudioDeviceGetProperty(devices[i], 0, false, kAudioDevicePropertyDeviceName, &size, device_name);
              ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:255:20: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
    OSStatus err = AudioDeviceGetProperty(id, 0, false, kAudioDevicePropertyDeviceUID, &size, &UIname);
                   ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:384:28: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
            OSStatus err = AudioDeviceGetProperty(inDevice, 0, kAudioDeviceSectionGlobal, kAudioDevicePropertyNominalSampleRate, &outSize, &tmp_sample_rate);
                           ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^
../macosx/coreaudio/JackCoreAudioDriver.mm:414:28: warning: 'AudioDeviceGetProperty' is deprecated: first deprecated in OS X 10.6 [-Wdeprecated-declarations]
            OSStatus err = AudioDeviceGetProperty(inDevice, 0, kAudioDeviceSectionGlobal, kAudioDevicePropertyBufferFrameSize, &outSize, &tmp_buffer_size);
                           ^
/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareDeprecated.h:818:1: note: 'AudioDeviceGetProperty' has been explicitly marked deprecated here
AudioDeviceGetProperty( AudioDeviceID           inDevice,
^

```

P.S. did you get my email about adding all the OSX apps from the JackOSX project to the jackaudio organisation? (I assumed you are the owner, or main maintainer judging from the CVS repo)

**Comment by lilrc** [comment]
Great work! In general, this looks good. Please read and respond to my comments.

**Comment by joshdekock** [comment]
@lilrc I've added a commit (2 now because I had a typo :( ) which responds to your comments

**Comment by joshdekock** [comment]
(Just reverted the commit which I accidentally pushed, sorry I'm not amazing at git yet :( )

**Comment by lilrc** [comment]
I'm confused. Why did you revert the "Updated som styling of wscripts" commit?

**Comment by lilrc** [comment]
If you want to get rid of the messy history you can just rebase and then force push.

**Comment by joshdekock** [comment]
Done.

**Comment by lilrc** [comment]
Now you didn't fix the
```
serverlib = bld(features = ['c', 'cxx', 'cxxshlib', 'cshlib'])
if bld.env['IS_MACOSX']:
    severlib.framework = ['CoreAudio', 'CoreFoundation', 'Accelerate']
```
thing and all its relatives. Or am I missing something?

File: common/wscript [hunk 1]
```diff
@@ -353,10 +363,19 @@ def build(bld):
         'JackAudioAdapterFactory.cpp',
          ]
 
-    if bld.env['BUILD_ADAPTER'] and bld.env['IS_MACOSX']:
-         audio_adapter_sources += ['../macosx/coreaudio/JackCoreAudioAdapter.cpp']
-         process = create_jack_process_obj(bld, 'audioadapter', audio_adapter_sources, serverlib)
-         process.env.append_value("LINKFLAGS", "-framework CoreAudio -framework AudioUnit -framework AudioToolbox -framework CoreServices")
+    if bld.env['BUILD_ADAPTER'] and bld.env['IS_MACOSX'] and False:
```
Comment (lilrc) [inline common/wscript]: Ok, so this does not work. Do you have an idea what might be wrong?

**Comment by sletz** [comment]
Why going for "mm" extension? What kind of new code would require objc++ compilation?

Le 15 d�c. 2015 � 21:13, Josh de Kock <notifications@github.com> a �crit :

**Comment by joshdekock** [comment]
@lilrc so it seems, I'm having a couple problems with git; and regarding the adapter, no idea, I haven't really looked into it yet. It complains about linking a lot, however if you link all the sources which it complains about it still complains about a source not being linked (even though it is being linked). I think I'll have to try manually compile to work out, how to do it, and what waf is doing wrong.

@sletz objc++ is so that compatibility with c++ is maintained, while also having access to objc-only functions, like the one to translate the OSStatus to a std::string.

**Comment by joshdekock** [comment]
@lilrc ok, I fixed the AudioAdapter, it was just not getting linked properly >:( Also, mach semaphores don't seem to work fully, posix ones look a lot better though. 

Can we merge this? It's not like it's going into master, only a WIP branch.

**Comment by lilrc** [comment]
Oh, sorry. Scratch that. You fixed it in the next commit. Maybe you should rebase those two "updated frameworks in wscripts" and "fixed frameworks in wscripts" into one.

**Comment by lilrc** [comment]
That last commit is fine for now. I will refactor stuff in common/wscript later on anyway. Just rebase those two commits I mentioned in the last comment and I'll merge this happily.

**Comment by joshdekock** [comment]
Hopefully, that's it.

**Comment by lilrc** [comment]
Thank you for your work!

**Pull request merged**